src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -560,6 +560,14 @@ private BoundExpression BindSimpleBinaryOperator(BinaryExpressionSyntax node, Di
 
             if (hasErrors)
             {
+                if (GetTupleCardinality(left).HasValue &&
+                    GetTupleCardinality(right).HasValue &&
```

Comment:
This does not appear to implement the specification-required overload resolution rules to determine which built-in operator to use. For example, if the left-hand operand is a tuple, and the right-hand operand is not a tuple but has a *conversion to* a tuple, we should be using tuple equality. Please add such a test.

Reply:
This may require LDM review/design of the intended rules if that isn't simple, which I suspect may be the case.

---
In reply to: [153581699](https://github.com/dotnet/roslyn/pull/23356#discussion_r153581699) [](ancestors = 153581699)

Reply:
See `TestEqualityOfTypeConvertingFromTuple` and `TestEqualityOfTypeConvertingToTuple`
The conversion to tuple doesn't contribute (so the binding errors out). 
The conversion from tuple works as before.

I believe that matches what you and I had discussed with Mads.